hub_updater.py

```diff
@@ -83,7 +83,33 @@ def get_pr_from_gh(pr_name, all_prs):
         return None
 
 
-def create_pr(manifest_path, requirements_path, module, jina_core_version, hub_repo, hub_origin, gh_hub_repo, all_prs) -> Optional[PullRequest]:
+def wrap_retry(func, param):
+    retry = True
+    time_wait = 0
+    while retry:
+        try:
+            time.sleep(time_wait)
+            return_obj = func(**param)
+            return return_obj
+        except GithubException as e:
+            if e.data \
+                    and isinstance(e.data, dict) \
+                    and 'message' in e.data.keys() \
+                    and 'Please retry your request again later' in e.data.get('message'):
+                print(f'Warning: Caught Github abuse detection mechanism exception')
+                retry = 'Retry-After' in return_obj.raw_headers.keys()
```

Comment:
it is quite weird to assume that the `returned_obj` exists.

Also, I do not see why don't we just check `retry` without doing all this weird if clauses? It will be very hard to mantain, the return messages can easily change

Reply:
Yeah, I haven't been able to reproduce it at all, so I'm not really sure what to expect. 

I have only tested it locally and on the hub-test duplicated repo. I'll clean it up once I understand it/reproduce it properly

Reply:
So I do not think we can assume this returned object exists if an exception has been raised. You can have a test wrapping a function which launches such an exception

Reply:
I would suggest just have a mechanism of retry N times after X seconds for every github action and wrap in a decorator and avoid all this noisy logic of understanding if that is the right exception